### PR TITLE
Language Server - Expose Location-related methods

### PR DESCRIPTION
**Pull request by fredg1**
Exposes and tests the methods that will be used to find the location of the various ParseTreeNodes.

### CODE REVIEW

File: src/net/sourceforge/kolmafia/textui/Parser.java [hunk 1]
```diff
@@ -3884,6 +3905,66 @@ private boolean atEndOfFile() {
...
+
+  private static Range makeInlineRange(final Position start, final int offset) {
+    Position end = new Position(start.getLine(), start.getCharacter() + offset);
+
+    return offset >= 0 ? new Range(start, end) : new Range(end, start);
+  }
+
+  private static Range mergeRanges(final Range start, final Range end) {
+    if (end == null
+        || start.getStart().getLine() > end.getEnd().getLine()
```
Comment (heeheehee-kolmafia) [inline src/net/sourceforge/kolmafia/textui/Parser.java:3926]: Can you instead use Positions.isBefore(end, start)?

File: src/net/sourceforge/kolmafia/textui/Parser.java [hunk 2]
```diff
@@ -3884,6 +3905,66 @@ private boolean atEndOfFile() {
...
+  private Location makeLocation(final Range range) {
+    String uri = this.fileUri != null ? this.fileUri.toString() : this.istream.toString();
+    return new Location(uri, range);
+  }
+
+  private static Location makeLocation(final Location start, final Range end) {
+    return Parser.mergeLocations(start, new Location(start.getUri(), end));
+  }
+
+  public static Location mergeLocations(final Location start, final Location end) {
```
Comment (heeheehee-kolmafia) [inline src/net/sourceforge/kolmafia/textui/Parser.java:3956]: Can you add a test for this method?

**Review by heeheehee-kolmafia — CHANGES_REQUESTED** [review]

File: src/net/sourceforge/kolmafia/textui/Parser.java (hunk 1 above)
Comment (fredg1) [reply src/net/sourceforge/kolmafia/textui/Parser.java:3926]: Well I'll be... I'm pleasantly surprised to learn of these utility methods...

Comment (heeheehee-kolmafia) [reply src/net/sourceforge/kolmafia/textui/Parser.java:3926]: As far as I can tell, it's just the one :(

https://github.com/eclipse/lsp4j/blob/main/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/util/Positions.java

Comment (fredg1) [reply src/net/sourceforge/kolmafia/textui/Parser.java]: well, this at least made me notice Ranges.containsRange and Ranges.containsPosition, which may come in handy in the future...

**Review by heeheehee-kolmafia — APPROVED** [review]

**Pull request merged**